### PR TITLE
add `use_aparam_as_mask` for pt backend

### PR DESCRIPTION
**Pull request by ChiahsinChu**
- allow `use_aparam_as_mask` for pt backend
- add test for dimension check

## Summary by CodeRabbit

- **New Features**
	- Enhanced configurability of the `InvarFitting` class with the addition of a `seed` parameter.
	- Introduced a new `use_aparam_as_mask` parameter affecting input dimension calculations and atomic parameter handling in both `InvarFitting` and `GeneralFitting` classes.

- **Bug Fixes**
	- Streamlined checks for the `use_aparam_as_mask` parameter to ensure proper error handling.

- **Tests**
	- Updated tests to incorporate the new `use_aparam_as_mask` parameter, including a new test case to validate its behavior.

### CODE REVIEW

**Comment by ChiahsinChu** [comment]
link to issue #4197

**Review by anyangml — APPROVED** [review]
LGTM

**Review by wanghan-iapcm — APPROVED** [review]

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
The implementation in the dp model is not covered by the unittest. 
I would suggest adding the [consistent UT test ](https://docs.deepmodeling.com/projects/deepmd/en/v3.0.0b4/development/create-a-model-pt.html#consistent-tests) between beckends (although the parameter should not have effect on output).

File: source/tests/consistent/fitting/test_ener.py [hunk 1]
```diff
@@ -42,7 +42,9 @@
     ("float64", "float32", "bfloat16"),  # precision
     (True, False),  # mixed_types
     (0, 1),  # numb_fparam
+    (0, 1),  # numb_aparam
     ([], [-12345.6, None]),  # atom_ener
+    (True, False),  # use_aparam_as_mask
```
Comment (njzjz) [inline source/tests/consistent/fitting/test_ener.py:47]: `numb_aparam == 0 and use_aparam_as_mask == True` can be skipped, which is not a meaningful parameter.

**Pull request closed without merging**